src/main/java/org/apache/cassandra/sidecar/restore/RestoreProcessor.java

```diff
@@ -312,13 +325,17 @@ void decrementActiveSliceCount(RestoreSlice slice)
         void captureImportQueueLength()
         {
             activeSliceCounterPerInstance.forEach((instanceId, counter) ->
-                                                  stats.captureSliceImportQueueLength(instanceId, counter.get()));
+                                                  metadataFetcher.instance(instanceId)
```

Comment:
should capture be specific to the instance we are processing at the moment? i.e. just pass the `slice` as parameter and report it for the individual slice?